packages/variable-editor/src/components/variables/master/OverwriteDialog.tsx

```diff
@@ -154,3 +137,30 @@ const toNodes = (root?: ProjectVarNode): Array<BrowserNode> => {
   }
   return root.children.map(varNode => toNode(varNode));
 };
+
+const addVariable = (variables: Variable[], node: ProjectVarNode): AddNodeReturnType<Variable> => {
+  const lastDot = node?.key.lastIndexOf('.');
+  const namespace = node?.key.substring(0, lastDot);
+  let metadataType: MetadataType = '';
+  if (isMetadataType(node.type)) {
+    metadataType = node.type;
+  }
+  const returnValue = addNode(node.name, namespace, variables, name => {
+    if (name === node?.name) {
+      return {
+        name,
+        value: node?.value,
+        children: [],
+        description: node?.description,
+        metadata: { type: metadataType }
+      };
+    }
+    return VariableFactory(name);
+  });
+  let childReturnValue = returnValue;
+  for (const child of node.children) {
+    childReturnValue = addVariable(childReturnValue.newData, child);
+  }
+  returnValue.newData = childReturnValue.newData;
+  return returnValue;
+};

```

Comment:
I think you can remove all the `?` question marks from `node?` since node will always be defined in here.